src/encrypt.rs

```diff
@@ -0,0 +1,464 @@
+//! COSE Encryption
+
+use openssl::rand::rand_bytes;
+use openssl::symm::{decrypt_aead, encrypt_aead, Cipher};
+use serde::{ser::SerializeSeq, Deserialize, Serialize, Serializer};
+use serde_bytes::ByteBuf;
+use serde_cbor::Error as CborError;
+use serde_cbor::Value as CborValue;
+
+use crate::error::COSEError;
+use crate::header_map::{map_to_empty_or_serialized, HeaderMap};
+
+#[allow(dead_code)]
+enum COSEAlgorithm {
+    /// AES-GCM mode w/ 128-bit key, 128-bit tag
+    A128GCM,
+    /// AES-GCM mode w/ 192-bit key, 128-bit tag
+    A192GCM,
+    /// AES-GCM mode w/ 256-bit key, 128-bit tag
+    A256GCM,
```

Comment:
I think it would be better to name these as
```
AesGcm96_128_128
AesGcm96_128_192
AesGcm96_128_256
```
It's a bit more consistent and readable, even if the spec says otherwise.
```
AesGcm128
AesGcm192
AesGcm256
```
is also acceptable, and mention in the comment that the nonce is 96 bits.